tests/select_for_update/tests.py

```diff
@@ -120,6 +120,27 @@ def test_for_update_sql_generated_of(self):
             expected = [value.upper() for value in expected]
         self.assertTrue(self.has_for_update_sql(ctx.captured_queries, of=expected))
 
+    @skipUnlessDBFeature('has_select_for_update_of')
+    def test_for_update_of_followed_by_values(self):
+        """
+        select_for_update(of=(...)) still works if the columns are restricted
+        by values/value_list.
+        """
+        with transaction.atomic():
+            list(iter(Person.objects.select_for_update(of=('self',)).values('pk')))
```

Comment:
the `iter` wrapping is unnecessary?

Reply:
Oops, made some experiments with `ModelIterable` (which is the only place which uses `klass_info` besides the FOR UPDATE OF code) and forgot to remove.